control-base/src/main/java/fi/nls/oskari/control/layer/GetLayerIds.java

```diff
@@ -30,11 +30,8 @@
     private static final Logger log = LogFactory.getLogger(GetLayerIds.class);
 
     private static final String LAYER_IDS = "layerIds";
-    private static final String ID = "id";
-
-    private static final List<Integer> extra_layers = new ArrayList<Integer>();
 
-    private static PermissionsService permissionsService = new PermissionsServiceIbatisImpl();
+    private static final List<Integer> extra_layers = new ArrayList<>();
```

Comment:
This list could be non-static since it's modified in the init() function. If only one instance of this class is ever created then it makes no difference whether or not the field is static (it gets executed just once anyway), but if there's a chance another instance gets created then this solution no longer works as intended.